### PR TITLE
feat: allow string value for `port`

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [X] This is a **feature**
- [ ] This is a **code refactor**
- [X] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### Motivation / Use-Case

Allow string for `client.webSocketURL.port`

### Breaking Changes
None

### Additional Info
No

### CODE REVIEW

File: test/validate-options.test.js [hunk 1]
```diff
@@ -149,9 +152,6 @@ const tests = {
       {
         webSocketURL: { host: '' },
       },
-      {
-        webSocketURL: { port: '' },
-      },
```
Comment (alexander-akait) [inline test/validate-options.test.js:154]: Keep it, should be fail when `minLength: 0` (in schema)

**Review by anshumanv — COMMENTED** [review]
Can we add a simple e2e so it doesn't fail during compilation?

**Comment by snitin315** [comment]
Added 👍🏻

**Review by anshumanv — DISMISSED** [review]

**Review by alexander-akait — COMMENTED** [review]
We also can have `port: string` and `webSocketServer: { type: 'ws', options: { port: string } }`, let's add these tests too, maybe will be great to move it in normalize and convert to `Number` to avoid working with different types:
```
if (typeof this.options.port === 'string' && this.options.port !== 'auto') {
  this.options.port = Number(this.options.port)
}

// Same for `websocketUrl.port` and `webSocketServer.options.port`
```

File: lib/utils/normalizeOptions.js [hunk 2]
```diff
@@ -91,6 +95,15 @@ function normalizeOptions(compiler, options, logger) {
...
     };
   } else {
+    if (
+      options.webSocketServer.options &&
+      typeof options.webSocketServer.options.port === 'string'
+    ) {
+      options.webSocketServer.options.port = Number(
+        options.webSocketServer.options.port
+      );
+    }
```
Comment (alexander-akait) [inline lib/utils/normalizeOptions.js:105]: Make sense to simplify it moving here https://github.com/webpack/webpack-dev-server/blob/5adf5d3a4de11451ff65c00d63c0d89d6f4e2ddf/lib/utils/normalizeOptions.js#L113 and use

```js
// i.e. normalize after normalize :)
if (typeof options.webSocketServer.port === 'string') {
  options.webSocketServer.port = Number(options.webSocketServer.port);
}
```

File: lib/utils/normalizeOptions.js [hunk 3]
```diff
@@ -111,9 +124,14 @@ function normalizeOptions(compiler, options, logger) {
 
     options.client.webSocketURL = {
       host: parsedURL.hostname,
-      port: parsedURL.port,
+      port: Number(parsedURL.port),
       path: parsedURL.pathname,
     };
+  } else if (
+    typeof options.client.webSocketURL === 'object' &&
```
Comment (alexander-akait) [inline lib/utils/normalizeOptions.js:131]: Wrong check and unnecessary, we always has `Object` here

Comment (alexander-akait) [reply lib/utils/normalizeOptions.js:131]: Just use `else`

**Review by alexander-akait — APPROVED** [review]
Feel free to merge when CI will be green

**Review by alexander-akait — APPROVED** [review]

**Review by anshumanv — APPROVED** [review]

**Pull request merged**